.taskcluster.yml

```diff
@@ -0,0 +1,70 @@
+version: 1
+policy:
+  pullRequests: public
+tasks:
+  $if: taskcluster_root_url == "https://taskcluster.internal.wpgapi.uk"
+  then:
+    $let:
+      imageName: gcr.io/wpg-internal/taskcluster-operator
+    in:
+      - $if: (tasks_for == "github-push") && (event.ref == "refs/heads/main")
+        then:
+          taskId:
+            $eval: as_slugid("push_task")
+          created: { $fromNow: '' }
+          deadline: { $fromNow: '3 hours' }
+          expires: { $fromNow: '10 days' }
+          provisionerId: shared
+          workerType: multistage-docker
+          scopes:
+            - secrets:get:shared/github
+            - secrets:get:shared/cloudbuild-service-account
+          routes:
+            - notify.slack-channel.C9Z43K24F.on-any
+          payload:
+            steps:
+              $flatten:
+                - image: gcr.io/wpg-internal/ci-bootstrap
+                - image: bitnami/git
+                  command: ["/bin/bash", "-c"]
+                  args:
+                    - git clone ${event.repository.html_url} . && git checkout ${event.after}
+                - image: docker
+                  args: [ "build", ".", "-t", "${imageName}:${event.after}" ]
+                - image: docker
+                  args: [ "push", "${imageName}:${event.after}" ]
+                - image: gcr.io/wpg-internal/update-infra-image
+                  args: ["deployments/packages/_taskcluster-operator/package.json=config.imageTags.taskcluster-operator=${event.after}"]
```

Comment:
what the hecking is this argument syntax

Reply:
let's see about noticing completed taskcluster tasks and having a service notice that and update infra for you by knowing the mapping from repos to images, and images to the packages that use them? :)